### PR TITLE
agent: Add runtime spec domainname support

### PR DESCRIPTION
**Pull request by dcantah**
Fixes: #5960

This change adds support across the stack for the newly added domainname field in the OCI runtime spec.

-Bumps runtime-spec to v1.0.3-0.20220909204839-494a5a6aca78 (main) 
-Updates oci.proto for the guest protocol and libs/oci for the rust equivalents.

### CODE REVIEW

**Comment by dcantah** [comment]
Will likely have time to take this for a better spin in a couple days

File: src/agent/rustjail/src/container.rs [hunk 1]
```diff
@@ -1297,6 +1303,31 @@ pub fn setup_child_logger(fd: RawFd, child_logger: Logger) -> tokio::task::JoinH
...
 }
 
+/// Sets domainname for process (see
+/// [setdomainname(2)](https://man7.org/linux/man-pages/man2/setdomainname.2.html)).
+///
+/// unistd doesn't have a safe binding for this, so need to make our own.
+/// Pulled from https://github.com/containers/youki/blob/6d698da2c70ca7bfa71699495157acf3d9930657/crates/libcontainer/src/syscall/linux.rs#L350-L369
+/// with some small tweaks.
+fn set_domainname(name: &str) -> Result<()> {
+    let ptr = name.as_bytes().as_ptr() as *const c_char;
```
Comment (mxpv) [inline src/agent/rustjail/src/container.rs:1313]: nit: `cast` is nicer: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a9631617aa5c467bda3bd289574ae77f

Comment (dcantah) [reply src/agent/rustjail/src/container.rs:1313]: TIL, will change!

Comment (dcantah) [reply src/agent/rustjail/src/container.rs]: Done

**Pull request closed without merging**